pkg/mutation/match/match.go

```diff
@@ -208,7 +208,7 @@ func namesMatch(match *Match, target *Matchable) (bool, error) {
 		return true, nil
 	}
 
-	return match.Name.Matches(target.Object.GetName()), nil
+	return match.Name.Matches(target.Object.GetName()) || match.Name.Matches(target.Object.GetGenerateName()), nil
```

Comment:
From this [doc](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/), GenerateName is a prefix, so only prefix-based name matching should apply.

Reply:
Any other type of name matching will not be reproducible, since the suffix will be random

Reply:
I am not following why only prefix-based name matching should apply. Could you provide some examples?

I think that if the wildcard is `*kube*` then any resources with generateName that contains `kube` should be matched right? ie matching both `kube-` and `abc-kube-` generateName with `*kube*` should result in `true`.

Reply:
Suffix matching would not be possible. Midfix might.

`generateName: "some-value"` -> `name: "some-value32454"`

`match.name: "some-value"` should not match, but would with this change

`match.name: "*some-value"` should not match, but would with this change

`match.name: "some-value*"` should match

`match.name: "*-value*"` should match

Reply:
Updated the code to add a new function specifically to match generate name